src/Hal.h

```diff
@@ -0,0 +1,118 @@
+/**@file
+ * This file is part of the TASTE Linux Runtime.
+ *
+ * @copyright 2025 N7 Space Sp. z o.o.
+ *
+ * TASTE Linux Runtime was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TASTE_HAL_H
+#define TASTE_HAL_H
+
+/**
+ * @file    Hal.h
+ * @brief   Header for Hal
+ */
+
+#include <cstdbool>
+#include <cstdint>
+#include <cstdlib>
+#include <mutex>
+
+#ifndef RT_MAX_HAL_SEMAPHORES
+#define RT_MAX_HAL_SEMAPHORES 8
+#endif
+
+namespace taste {
+
+class Hal final
```

Comment:
The point for Hal was to have a common API that can be used by user functions to access functionalities such as time and semaphores, on Leon, Linux, and future SAMX runtimes, so that there can be a re-usable component, like a TimeProvider, that can work on each of them. 
The task explicitly stated:
"Implement HAL for TASTE Linux Runtime, in line with MBEP-N7S-RT-SDD"
So, I expect the signatures to be compatible with C code, via extern C.

Reply:
Internally, C++ can be used -> the intent is for the signatures to be usable in C code, and exchangeable with what is available on Leon and future SAMX